### PR TITLE
#7111 Fix deactivation/delete actions for unsaved standalone mod components

### LINKED ISSUE #7111: Confusing Deactivation message for Standalone Mod Components
## Describe the bug

If you've never saved a Standalone Mod Component, deactivating it will actually delete it. This conflicts with the message in the popover suggesting reactivation in the Extension Console.

## To Reproduce

1. Create a standalone mod component
2. Do _not_ save it
3. Deactivate it

## Actual behavior

The "Deactivate Mod?" modal appears telling developers they can reactivate the mod in the Extension Console

## Expected behavior

The modal should inform developers that the unsaved Standalone Mod Component will be deleted and be unable to be recovered.

## Desktop (please complete the following information):

- Extension Version 1.8.5-alpha.3

## Additional context

https://pixiebrix.slack.com/archives/C0436P48QHY/p1702435676810429

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Fixes #7111 

## Discussion

- Please see demo for discussion on future work

## Demo

- https://www.loom.com/share/a75cd5fd464b4e34a4f5f5cf0f17e73a

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer @grahamlangford

### CODE REVIEW

**Review by grahamlangford — APPROVED** [review]

File: src/pageEditor/hooks/useRemoveModComponentFromStorage.tsx [hunk 1]
```diff
@@ -31,24 +34,63 @@ import { removeExtensionsFromAllTabs } from "@/store/uninstallUtils";
 
 type Config = {
   extensionId: UUID;
-  shouldShowConfirmation?: boolean;
+  confirmationModal?: ConfirmationModalProps;
```
Comment (BLoe) [inline src/pageEditor/hooks/useRemoveModComponentFromStorage.tsx:37]: Please add a jsDoc comment somewhere, either on this type or the hook or something, that explains how the "optional modal props --> do we show confirm modal?" logic works. I saw you removed the explicit `shouldShowConfirmation` flag, and it took me a minute to understand the changes.

**Review by BLoe — APPROVED** [review]
Just a couple nits, looks good though. Nice work on the selector-with-an-input, those can be tricky to set up!

File: src/pageEditor/hooks/useRemoveModComponentFromStorage.tsx (hunk 1 above)
Comment (mnholtz) [reply src/pageEditor/hooks/useRemoveModComponentFromStorage.tsx]: Would changing `confirmationModal?: ConfirmationModalProps` to `showConfirmationModal?: ConfirmationModalProps` make more sense?

Comment (mnholtz) [reply src/pageEditor/hooks/useRemoveModComponentFromStorage.tsx]: I also added a comment

**Pull request merged**